Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/handlers_test.go

```diff
@@ -41,13 +42,21 @@ func (f fakeRL) CanAccept() bool { return bool(f) }
 func (f fakeRL) Accept()         {}
 
 func expectHTTP(url string, code int, t *testing.T) {
-	r, err := http.Get(url)
+	var response *http.Response
+	err := wait.Poll(10, time.Duration(100*time.Millisecond), func() (bool, error) {
+		var e error
+		response, e = http.Get(url)
+		if response.StatusCode == 429 {
```

Comment:
Yeah, this isn't the correct fix.  Based on reading the test, he should have had `expectHTTP` take a channel to indicate completion of the request and guarantee synchronization of multiple blocked requests.  That would allow him to avoid the failure he's hitting now where the block is released, but the queue doesn't get cleared fast enough.